### PR TITLE
51 implement sing in user functionality rb

### PR DESCRIPTION
**Pull request by RajaaBoulassouak**
#### Description
- Connect SignInUser component to Redux Store.
- Create function for updating inputs.
- Implement function that takes input values.
- Add tests for SignInUser component functions
- Add 'New User?' to Sign In page for directing to Create Account form (references #47)
----
#### Issues # and Action
- references #51 
----
#### TO DO / Noteworthy
- Step after successfully signing in not determined yet. 
----
#### Features
- [x] New Feature
- [ ] Expand Feature
- [ ] Optimize/Improve Feature

----
#### Testing
- [ ] No Testing
- [ ] Some Testing
- [x] Sufficient Testing

- [ ] Left notes near functions that have not been tested
- [ ] Intentionally skipped / failing / empty tests as a reminder to add/improve these later

- [ ] Tests have been changed
- [ ] Existing tests are now failing / skipped

----
#### Deployment
- [ ] This is a complete feature, ready for deployment
- [x] This is a partial feature, NOT ready for deployment

----
#### API Changes
(please detail in description)
- [ ] New API
- [ ] Existing API, New Endpoint
- [ ] README.md is updated

----
#### Tech Changes
(please describe)
- [ ] New Functional Library
- [ ] New Testing Library
- [ ] New Debugging Tool
- [ ] README.md is updated

### CODE REVIEW

File: src/SignInUser/SignInUser.test.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  let mockUserName; 
+  let mockPassword;
+  let mockEvent
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = shallow(<SignInUser signIn={jest.fn()}/>)
+    mockUserName = 'Rajaa'; 
+    mockPassword = 'password'; 
+    mockEvent = {
```
Comment (RajaaBoulassouak) [inline src/SignInUser/SignInUser.test.js]: Not sure if this mockEvent is supposed to include this or token: 

mockEvent = {
     target: {
       name: 'token',
       value: 'token12345'
      }
    }

**Comment by RajaaBoulassouak** [comment]
Hi Derek. This is what I got so far. After clicking Sign In not sure if everything is working and user will be signed in successfully, since we didn't specify what will happen next. Also in the src/SignInUser/SignInUser.test.js on line 21-26  and  41-46. Not sure if the target should be 'UserName' and a value or 'token' and a value like this: 
mockEvent = {
     target: {
       name: 'token',
       value: 'token12345'
      }
    }

File: src/SignInUser/SignInUser.test.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     target: {
+       name: 'userName',
+       value: 'Derek'
+      }
+    }
+  })
+
+  it('should match snapshot', () => {
+    const wrapper = shallow(<SignInUser />)
+
```
Comment (drhanciles) [inline src/SignInUser/SignInUser.test.js]: we can delete this line. we don't need it because we are assigning wrapper on line 13

**Comment by drhanciles** [comment]
> Hi Derek. This is what I got so far. After clicking Sign In not sure if everything is working and user will be signed in successfully, since we didn't specify what will happen next. Also in the src/SignInUser/SignInUser.test.js on line 21-26 and 41-46. Not sure if the target should be 'UserName' and a value or 'token' and a value like this:
> mockEvent = {
> target: {
> name: 'token',
> value: 'token12345'
> }
> }
The mock event needs to have username, not token because token is what we get back from request and is being stored in state. The mock event only involves the username or password inputs.

**Comment by RajaaBoulassouak** [comment]
Ok, Thanks! The change you suggested has been updated.

**Comment by drhanciles** [comment]
Great work today Rajaa! I'll get this merged into master.

**Pull request merged**